components/nexus-orient/src/main/java/org/sonatype/nexus/orient/entity/EntityAdapter.java

```diff
@@ -259,33 +295,28 @@ protected void setMetadata(final T entity, final ODocument doc) {
    * Get metadata from entity.
    */
   protected EntityMetadata getMetadata(final T entity) {
-    checkNotNull(entity);
-    EntityMetadata metadata = entity.getEntityMetadata();
-    checkState(metadata != null);
-    return metadata;
+    return metadata(entity);
+  }
+
+  public EntityId encode(final ORID id) {
+    return new EntityId(getRecordIdObfuscator().encode(getType(), id));
+  }
+
+  public ORID decode(final EntityId id) {
+    return getRecordIdObfuscator().decode(getType(), id.toString());
   }
 
   /**
    * Return record identity of entity.
    */
-  protected ORID recordIdentity(final T entity) {
-    EntityMetadata metadata = getMetadata(entity);
-    if (metadata instanceof AttachedEntityMetadata) {
-      return ((AttachedEntityMetadata) metadata).rid;
-    }
-    return getRecordIdObfuscator().decode(getType(), metadata.getId().getValue());
+  public ORID recordIdentity(final T entity) {
```

Comment:
Not directly related to the PR since this name was pre-existing, but anyone else think this should be prefixed with 'get' so it's clear that it's not an action? When I first saw the calls it was not clear this was a getter and not performing an action